Assignment 1- Calculator using BODMAS Preference in C

KoushikNagaraja opened the pull request:
This is an C based calculator program that takes user input as a string and performs BODMAS Operations

KoushikNagaraja committed:
Add files via upload

This is an C based calculator program that takes user input as a string and performs BODMAS Operations

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+#include<string.h>
+#include<ctype.h>
+#include<stdlib.h>
+int calculate(int a,int b,char exp)
```
apply code formatting and match function names with this standards, take care of this at all places.

Use camelcase for function names.
Start function names with a verb to indicate action.
Keep names concise but meaningful.
ex :
int calculateSum(int a, int b);
void displayMessage(const char *message);

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+}
+int evaluateexpression(const char *character)
+{
+    int nums[100];
```
Can variable names match below standards, take care of this at all places: 

- Use descriptive names: Choose meaningful names that reflect the purpose of the variable.
- Use Camelcase: Use lowercase letters with underscores to separate words.
ex : 
int totalCount;
float averageScore;
char studentName[50];


Global variables:
ex :
int totalCount;

Constants:
- Use all uppercase letters with underscores to separate words.

ex:
#define MAX_BUFFER_SIZE 1024
const float PI = 3.14159

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+    char operators[100];
+    int numsindex=0;
+    int operatorindex=0;
+    int i=0;
```
Can you use precise names than i, j, k should be like iteratorI, iteratorJ, iteratorK for code readability.

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+        default:printf("Invalid operator");exit(0);
+    }
+}
+int evaluateexpression(const char *character)
```
Can you split this multiple function doing specific task as this function is doing multiple tasks?

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+        if(isdigit(character[i]))
+        {
+            int value=0;
+            while(isdigit(character[i]))
```
Avoid using inbuilt functions unless it is system calls, please change the built-in functions to user implementation function. take care at all places.

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+            }
+            nums[numsindex++]=value;
+        }
+        else if(strchr("+-*/",character[i]))
```
rather than hardcoding the value "*/+-" at all places, you can add as macro and use the same at all other places.

take care of hardcoded values.

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+        }
+        else
+        {
+            printf("invalid character in expression");exit(0);
```
can you print the expected the error string asked in problem statement,

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+        case '-':return a-b;
+        case '*':return a*b;
+        case '/':if(b==0){printf("division by zero error");exit(0);}return a/b;
+        default:printf("Invalid operator");exit(0);
```
can you print the expected the error string asked in problem statement, both for invalid operator and division by zero error as well

gnanesh-c commented on calculator.c:
```diff
@@ -0,0 +1,81 @@
+    int result=evaluateexpression(input);
+    printf("%d",result);
+    return 0;
+}
```
add new line to clear this

KoushikNagaraja committed:
Update calculator.c

expanded one function to different functions and used proper naming conventions
